etl/transformer/mainnet_impl.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/block-explorer/blob/master/LICENSE.md.
+
+package transformer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/insolar/block-explorer/etl/types"
+)
+
+type MainNetTransformer struct {
+	stopSignal      chan bool
+	extractorChan   <-chan *types.PlatformJetDrops
+	transformerChan chan *types.JetDrop
+}
+
+func NewMainNetTransformer(ch <-chan *types.PlatformJetDrops) *MainNetTransformer {
+	return &MainNetTransformer{
+		stopSignal:      make(chan bool, 1),
+		extractorChan:   ch,
+		transformerChan: make(chan *types.JetDrop),
+	}
+}
+
+func (m *MainNetTransformer) Start(ctx context.Context) error {
+	// belogger.FromContext(ctx).Info("MainNetTransformer is starting")
+	go func() {
+		defer func() { _ = m.Stop(ctx) }()
+		for {
+			m.run()
+			if m.needStop() {
+				return
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (m *MainNetTransformer) Stop(ctx context.Context) error {
+	ctx.Done()
+	m.stopSignal <- true
+	return nil
+}
+
+func (m *MainNetTransformer) GetJetDropsChannel() <-chan *types.JetDrop {
+	return m.transformerChan
+}
+
+func (m *MainNetTransformer) needStop() bool {
+	select {
+	case <-m.stopSignal:
+		return true
+	default:
+		// continue
+	}
+	return false
+}
+
+func (m *MainNetTransformer) run() {
+	if m.needStop() {
```

Comment:
With `Stop` you send one msg to `stopSignal` channel. If you catch it here, you will return to line 35 (`for` loop at `Start`), and second call to `m.needStop()` return false, and loop will not ended.